presto-docs/src/main/sphinx/develop/worker-protocol.rst

```diff
@@ -125,3 +125,18 @@ upstream workers using buffer number 2.
 
 .. image:: worker-protocol-output-buffers.png
   :width: 600
+
+Failure Handling
+~~~~~~~~~~~~~~~~
+
+Task failures are reported to the coordinator via TaskStatus and TaskInfo updates.
+
+When a task failure is discovered, the coordinator aborts all remaining tasks and
+reports a query failure to the client. When a task failure occurs (or an abort
+request is received), all further processing stops, and all remaining task output
+is discarded.
+
+Failed/aborted tasks continue responding to data plane requests as normal to prevent
```

Comment:
> continue responding to data plane requests as normal

perhaps, clarify that the responses are empty.

Reply:
Also mentioned `X-Presto-Buffer-Complete` is equal to `false` to prevent downstream tasks from succeeding